src/psyclone/domain/nemo/transformations/nemo_arrayrange2loop_trans.py

```diff
@@ -0,0 +1,275 @@
+            # place it within a NemoKern (inlined kernel) node.
+            parent = assignment.parent
+            # We do not provide the fparser2 ast of the code as we are
+            # moving towards using visitors rather than gen_code when
+            # outputting nemo api code
+            inlined_kernel = NemoKern([assignment], None, parent=parent)
+            parent.children = [inlined_kernel]
+
+    def __str__(self):
+        return (
+            "Convert the PSyIR assignment for a specified ArrayReference "
+            "Range into a PSyIR NemoLoop.")
+
+    @property
+    def name(self):
+        '''
+        :returns: the name of the transformation as a string.
+        :rtype: str
+
+        '''
+        return type(self).__name__
+
+    def validate(self, node, options=None):
+        '''Perform various checks to ensure that it is valid to apply the
+        NemoArrayRange2LoopTrans transformation to the supplied PSyIR Node.
+
+        :param node: the node that is being checked.
+        :type node: :py:class:`psyclone.psyir.nodes.Range`
+        :param options: a dictionary with options for \
+            transformations. No options are used in this \
+            transformation. This is an optional argument that defaults \
+            to None.
+        :type options: dictionary of string:values or None
+
+        :raises TransformationError: if the node argument is not a \
+            Range, if the Range node is not part of an ArrayReference, \
+            if the Range node is not the outermost Range node of the \
+            ArrayReference or if that ArrayReference does not \
+            consitute the left hand side of an Assignment node.
+
+        '''
+        # Am I Range node?
+        if not isinstance(node, Range):
+            raise TransformationError(
+                "Error in NemoArrayRange2LoopTrans transformation. The "
+                "supplied node argument should be a PSyIR Range, but "
+                "found '{0}'.".format(type(node).__name__))
+        # Am I within an array reference?
+        if not node.parent or not isinstance(node.parent, ArrayReference):
+            raise TransformationError(
+                "Error in NemoArrayRange2LoopTrans transformation. The "
+                "supplied node argument should be within an ArrayReference "
+                "node, but found '{0}'.".format(type(node.parent).__name__))
+        array_ref = node.parent
+        # Is the array reference within an assignment?
+        if not array_ref.parent or not isinstance(array_ref.parent,
+                                                  Assignment):
+            raise TransformationError(
+                "Error in NemoArrayRange2LoopTrans transformation. The "
+                "supplied node argument should be within an ArrayReference "
+                "node that is within an Assignment node, but found '{0}'."
+                .format(type(array_ref.parent).__name__))
```

Comment:
It will return None if parent is None which I think is what it should do. Test `test_within_assignment` in `nemo_arrayrange2loop_trans_test.py` checks for this.

Reply:
Oh yes, ignore me.